File: ABC/abc017/abc017_3 1.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 30点解法
    int N, M, ans = 0;
    cin >> N >> M;

    vector<int> l(N), r(N), s(N);
    for (int i = 0; i < N; i++) cin >> l[i] >> r[i] >> s[i];

    for (int b = 0; b < (1 << N); b++) {
        int S = 0;
        vector<bool> jewels(M, false);
        for (int i = 0; i < N; i++) {
            if ((b >> i) & 1) {
                S += s[i];
                for (int j = l[i] - 1; j < r[i]; j++) jewels[j] = true;
            }
        }

        for (int j = 0; j < M; j++) {
            if (jewels[j] == false) { // 獲得していない宝石がある
                ans = max(ans, S);
                break;
            }
        }
    }

    cout << ans << endl;
}

File: ABC/abc017/abc017_3 1.py
#30点解法
N, M = map(int, input().split())

l = [0] * N
r = [0] * N
s = [0] * N
for i in range(N):
    l[i], r[i], s[i] = map(int, input().split())

ans = 0
for b in range(1 << N):
    S = 0
    jewels = [False] * M
    for i in range(N):
        if (b >> i) & 1:
            S += s[i]
            for j in range(l[i] - 1, r[i]):
                jewels[j] = True

    for j in range(M):
        if jewels[j] == False: #獲得していない宝石がある
            ans = max(ans, S)
            break

print(ans)